### ISSUE TITLE
PEP 487

### ISSUE THREAD

**Issue by afg984** [PyCQA/pylint#1620]
### Steps to reproduce
Run pylint against the following source code:
```python
class IntField:
    def __get__(self, instance, owner):
        return instance.__dict__[self.name]

    def __set__(self, instance, value):
        if not isinstance(value, int):
            raise ValueError(f'expecting integer in {self.name}')
        instance.__dict__[self.name] = value

    # this is the new initializer:
    def __set_name__(self, owner, name):
        self.name = name
```

### Current behavior
```
No config file found, using default configuration
************* Module pep487
C:  1, 0: Missing module docstring (missing-docstring)
C:  1, 0: Missing class docstring (missing-docstring)
W: 11,27: Unused argument 'owner' (unused-argument)
R:  1, 0: Too few public methods (0/2) (too-few-public-methods)
W: 12, 8: Attribute 'name' defined outside __init__ (attribute-defined-outside-init)

------------------------------------------------------------------
Your code has been rated at 4.44/10 (previous run: 4.44/10, +0.00)
```

### Expected behavior
I believe these two should not be shown
```
W: 11,27: Unused argument 'owner' (unused-argument)
W: 12, 8: Attribute 'name' defined outside __init__ (attribute-defined-outside-init)
```

### pylint --version output
```
No config file found, using default configuration
pylint 1.8.0, 
astroid 1.5.3
Python 3.6.2 (default, Jul 20 2017, 03:52:27) 
[GCC 7.1.1 20170630]
```
(pylint is installed via `pip install git+https://github.com/PyCQA/pylint`)

### Reference
https://docs.python.org/3/whatsnew/3.6.html#pep-487-descriptor-protocol-enhancements

-- for the sample code and description of PEP 487

**Comment by PCManticore** [comment]
``unused-argument`` is not emitted any longer. For ``attribute-defined-outside-init`` I'm not so sure we shouldn't emit, it would be a good practice to just initialize it to ``None`` in an ``__init__`` method.

**Comment by afg984** [comment]
For `attribute-defined-outside-init`:
I think's it's not common for `__get__` being called before `__set_name__`, as `__set_name__` is called when the owner class of the descriptor object is created, covering both getting the attribute from the owner class, as well as from an instance of that class.

Thanks for fixing it in such a short time :+1:
